### PR TITLE
Chapter 13

### PR DESCRIPTION
**Pull request by balqis277**
ready

### CODE REVIEW

**Comment by balqis277** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -93,7 +93,7 @@ input {
 }
 
 input.btn.btn-primary {
-  margin-top: 10px;
+    margin-top: 10px;
```
Comment (anlvq-1058) [inline app/assets/stylesheets/custom.scss:96]: worng indent

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -2,13 +2,15 @@ class UsersController < ApplicationController
   before_action :logged_in_user, except: %i(new show create)
   before_action :correct_user, only: %i(edit update)
   before_action :admin_user, only: :destroy
+  protect_from_forgery
```
Comment (anlvq-1058) [inline app/controllers/users_controller.rb:5]: why you need this line

File: app/models/micropost.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+  scope :recent_posts, -> {order created_at: :desc}
```
Comment (anlvq-1058) [inline app/models/micropost.rb:3]: where the scope used

File: app/views/static_pages/home.html.erb [hunk 4]
```diff
@@ -1,8 +1,25 @@
...
+      <section class="micropost_form">
+        <%= render "shared/micropost_form" %>
+      </section>
+    </aside>
+    <div class="col-md-8">
+      <h3><%= "Micropost Feed" %></h3>
+      <%= render "shared/feed" %>
+    </div>
+  </div>
+<% else %>
```
Comment (anlvq-1058) [inline app/views/static_pages/home.html.erb:24]: indent please

**Review by anlvq-1058 — COMMENTED** [review]
cmt

File: app/views/static_pages/home.html.erb (hunk 4 above)
Comment (balqis277) [reply app/views/static_pages/home.html.erb:24]: I am sorry but I still could not detect where is the indent error. I already compared with the slides and few coding of my friends, still I could not find the indent error.

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: app/views/shared/_error_messages.html.erb [hunk 5]
```diff
@@ -1,13 +1,12 @@
 <% if object.errors.any? %>
-  <div id="error_explanation">
-    <div class="alert alert-danger">
-      The form contains
-      <%= pluralize(@user.errors.count, "error") %>.
+    <div id="error_explanation">
+      <div class="alert alert-danger">
+        The form contains
+        <%= pluralize(object.errors.count, "error") %>.
```
Comment (anlvq-1058) [inline app/views/shared/_error_messages.html.erb:5]: missing close tag div for `<div class="alert alert-danger">`

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

File: app/views/shared/_error_messages.html.erb [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
+      <ul>
```
Comment (huongnt-2545) [inline app/views/shared/_error_messages.html.erb:6]: Line 6 -> 11
```suggestion
      </div>
      <ul>
        <% object.errors.full_messages.each do |msg| %>
          <li><%= msg %></li>
        <% end %>
      </ul>
```

File: app/views/static_pages/home.html.erb [hunk 7]
```diff
@@ -1,8 +1,25 @@
...
- This is the home page for the
- <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
- sample application.
-</p
+  This is the home page for the
+  <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
+  sample application.
+</p>
+
+  <% if logged_in? %>
```
Comment (huongnt-2545) [inline app/views/static_pages/home.html.erb:9]: wrong indent from line 9 -> 24
```suggestion
<% if logged_in? %>
  <div class="row">
    <aside class="col-md-4">
    ...
```

**Comment by balqis277** [comment]
ready

**Comment by balqis277** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged
You can merge this pull

**Pull request merged**